src/qtcore/qml/qml.js

```diff
@@ -209,23 +209,32 @@ function construct(meta) {
     return item;
 }
 
+const typeDefaultValues = {
+  int: 0,
+  real: 0,
+  double: 0,
+  string: '',
+  bool: false,
+  list: [],
+  url: ''
+};
+
 /**
  * Create property getters and setters for object.
  * @param {Object} obj Object for which gsetters will be set
  * @param {String} propName Property name
  * @param {Object} [options] Options that allow finetuning of the property
  */
-function createSimpleProperty(type, obj, propName, options) {
+function createProperty(type, obj, propName, options = {}) {
     var prop = new QMLProperty(type, obj, propName);
     var getter, setter;
 
-    if (typeof options == 'undefined')
-      options = {};
-    else if (typeof options != 'object')
-      options = { default: options }
-
     obj[propName + "Changed"] = prop.changed;
     obj.$properties[propName] = prop;
+    if (!options.hasOwnProperty('default')) {
```

Comment:
maybe we can go for the `obj.$properties[propName].set(options.default)` variant, in there we differentiate between those cases anyway, in case of QtObjects we can set it to `null` and else to a default constructed value (it even makes some sense to set the value to null if assigned undefined from js, although Qt throws an error in that case)

Btw. the call to set should contain a reason, but that might be unrelated.